src/test/compile-fail/macros-nonfatal-errors.rs

```diff
@@ -14,14 +14,14 @@
 #![feature(asm)]
 #![feature(trace_macros, concat_idents)]
 
-#[derive(Zero)] //~ ERROR
+#[derive(Zero)]
 struct CantDeriveThis;
```

Comment:
I believe it's because we abort [here](https://github.com/rust-lang/rust/blob/master/src/librustc_driver/driver.rs#L691-L693) due to other errors during expansion, so we never get to `finalize_current_module_macro_resolutions`.